lib/rex/random_identifier/generator.rb

```diff
@@ -62,10 +62,42 @@ class ExhaustedSpaceError < StandardError; end
     ).uniq.freeze
   )
 
+  JavaScriptOpts = JavaOpts.merge(
+    forbidden: (
+      JavaOpts[:forbidden] +
+        # JavaScript Reserved Words + JavaScript Objects, Properties, and Methods + Java Reserved Words + Other Reserved Words + HTML Event Handlers
+        # https://www.w3schools.com/js/js_reserved.asp
```

Comment:
Looks like your list is misisng keywords like `const`, `class` etc; it's also missing identifiers with special meanings - such as `async`. I'd probably recommend MDN:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#reserved_words
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#identifiers_with_special_meanings
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#future_reserved_words

Or the specification:

https://tc39.es/ecma262/#sec-keywords-and-reserved-words

Reply:
I wasn't sure if this should be an explicit list or not - I had removed `const`, `class` and all those already defined in `JavaOpts[:forbidden]` as it adds those in with:
```
    forbidden: (
      JavaOpts[:forbidden] +
```

 I've added them back in and updated the list to comply with this words listed in the URLs above. Thank you!

Reply:
```
    forbidden: (
      JavaOpts[:forbidden] +
```

Ah I didn't spot that on the first read, I Wouldn't have expected JavaScript's word list to extend Java's (https://en.wikipedia.org/wiki/Principle_of_least_astonishment 😄) but that's all good - apologies :+1:

Reply:
I think this updated list is still missing the inbuilt window/global instance methods like btoa etc:

https://github.com/rapid7/rex-random_identifier/pull/12#discussion_r1840804575

Which folk may be reaching for in their payloads

```
$n ode
Welcome to Node.js v22.6.0.
Type ".help" for more information.
> btoa
[Function: btoa]
```